src/backend/opengl/opengl_display.cpp

```diff
@@ -201,23 +204,21 @@ void c_OpenGLDisplay::OnPaint(wxPaintEvent&)
 
         gl::BindProgramTextures(prog, { {&m_Processor->GetOriginalImg(), uniforms::Image} });
 
-        const auto scrollpos = m_ImgView.GetScrollPosition();
-        prog.SetUniform2i(uniforms::ScrollPos, scrollpos.x, scrollpos.y);
-        prog.SetUniform2i(
-            uniforms::ViewportSize,
-            m_GLCanvas->GetSize().GetWidth(),
-            m_GLCanvas->GetSize().GetHeight()
-        );
+        const auto scrollPos = m_ImgView.GetScrollPosition();
+        prog.SetUniform2i(uniforms::ScrollPos, scrollPos.x, scrollPos.y);
+        prog.SetUniform2i(uniforms::ViewportSize,
+                          canvasSize.GetWidth() * m_ScaleFactor,
+                          canvasSize.GetHeight() * m_ScaleFactor);
 
         m_VBOs.wholeImgScaled.Bind();
         gl::SpecifyVertexAttribPointers();
         glDrawArrays(GL_TRIANGLE_FAN, 0, 4);
 
         RenderProcessingResults();
         MarkSelection();
-    }
 
-    m_GLCanvas->SwapBuffers();
+        m_GLCanvas->SwapBuffers();
```

Comment:
This needs to be unconditional; otherwise on Linux there are random screen buffer fragments shown.

Reply:
I see. It asserts/abort() if GL Context is not assigned. But if we assign GL Context on macOS on each OnPaint then the assert goes away.